brave/src/main/java/brave/handler/MutableSpan.java

```diff
@@ -883,6 +928,32 @@ static void writeEndpoint(StringBuilder b,
     b.append('}');
   }
 
+  static Object[] add(Object[] input, Object key, Object value) {
+    int newIndex = input.length;
+    Object[] result = Arrays.copyOf(input, input.length + 2); // copy-on-write
+    result[newIndex] = key;
+    result[newIndex + 1] = value;
+    return result;
+  }
+
+  // this is externally synchronized, so we can edit it directly
+  static Object[] update(Object[] input, int i, Object value) {
+    if (value.equals(input[i + 1])) return input;
+    input[i + 1] = value;
+    return input;
+  }
+
+  static Object[] copy(Object[] input) {
+    return input.length > 0 ? Arrays.copyOf(input, input.length) : EMPTY_ARRAY;
+  }
+
+  static Object[] remove(Object[] input, int i) {
+    if (input.length == 2 && i == 0) return EMPTY_ARRAY;
+    Object[] result = Arrays.copyOf(input, input.length - 2);
+    System.arraycopy(input, i + 2, result, i, input.length - i - 2);
```

Comment:
oops we don't need to do this.. all we need to do is shift and set null. Otherwise dropping all but one tag (like skeletal spans do) will thrash. will fix